gadget/mountedfilesystem.go

```diff
@@ -801,6 +860,38 @@ func (f *mountedFilesystemUpdater) Rollback() error {
 		return fmt.Errorf("cannot map preserve entries for mount location %q: %v", f.mountPoint, err)
 	}
 
+	knownContent := make(map[string]bool)
+
+	for _, c := range f.ps.ResolvedContent {
+		knownContent[c.Target] = true
+	}
+
+	if f.fromPs != nil {
+		for _, c := range f.fromPs.VolumeStructure.Content {
+			if knownContent[c.Target] {
+				continue
+			}
+
+			destPath, backupPath := f.entryDestPaths(f.mountPoint, "", c.Target, backupRoot)
+
+			if osutil.IsDirectory(destPath) {
+				continue
+			}
+
+			if err := os.Remove(destPath); err != nil {
+				if !os.IsNotExist(err) {
```

Comment:
Do we really want to return error here? This method possibly need to be idempotent if called from a task (case of unexpected reboot, crash, etc. at this point should be handled).

Reply:
This condition is to make the function idempotent. So if the file was removed already, then we will not fail. Or did I misunderstand?

Reply:
Oh, sorry, was reading this the other way around, ignore the comment